theme/src/components/TableOfContents/GlobalTableOfContents.tsx

```diff
@@ -0,0 +1,207 @@
+import { Collapse } from '@material-ui/core';
+import clsx from 'clsx';
+import { useRouter } from 'next/router';
+import { createElement, Fragment } from 'react';
+
+import { ExternalLink } from '@/components/icons';
+import { Link } from '@/components/Link';
+import { TOCHeader } from '@/context/jupyterBook';
+import { isExternalUrl } from '@/utils/url';
+
+import styles from './GlobalTableOfContents.module.scss';
+
+interface Props {
+  headers: Record<string, TOCHeader>;
+  rootHeaders: string[];
+}
+
+/**
+ * Enum values for identifying header levels in the global table of contents
+ * data structure.
+ */
+enum Header {
+  Category = 1,
+  Title = 2,
+  Subtitle = 3,
+}
+
+/**
+ * Constant for applying styles to the title headers.
+ */
+const HEADER_TITLES = [Header.Title, Header.Subtitle];
+
+/**
+ * Component for rendering the global table of contents list. This uses a
+ * recursive render function to render the global TOC data structure since the
+ * data is deeply nested.
+ */
+export function GlobalTableOfContents({ headers, rootHeaders }: Props) {
+  const router = useRouter();
+
+  /**
+   * @returns The router pathname without hashes or query parameters.
+   */
+  function getPathname() {
+    return new URL(router.asPath, 'http://tmp.com').pathname;
```

Comment:
clean up 'http://tmp.com'

Reply:
I'm not sure what you mean specifically. Should we change the name? The `'http://tmp.com'` is necessary because the `URL` constructor requires a base if the first argument is just a path (which it is because we use `router.asPath`)

Reply:
The name though is arbitrary because it's not even used since we use the value `URL(...).pathname)`

Reply:
I see. Thanks for explaining. Could you add comments in the code for what you explained here as it's a little confusing.
Wouldn't router.asPath end up being the same as the pathname return value anyway?